### DISCUSSION TITLE
What is lacking to allow windows or even the Deck itself as a dev env?

### DISCUSSION THREAD

**Discussion by LuisMayo** [SteamDeckHomebrew/decky-loader#135] [Q&A] [answered]
After reading the docs I think I kind of understand how the dev environment works.

You need a Arch Linux (or similar) system. Then you either run "nodeck.sh" to develop locally or "deck.sh" to develop using Steam Deck as running env.

Assuming my knowledge is correct.
Why can't windows be used to develop, what points does the team think are not compatible?
Similarly, why couldn't the deck itself be used as a dev environment?

Thanks and sorry if they're too broad questions.

**Comment by TrainDoctor** [comment, marked as answer]
The Steam Deck is a linux machine. Developing for it on windows while possible is not the best idea.
Also those scripts install uncompilied python versions that were helpful during development of the react update but probably need to be updated to provide a binary install instead.
Using the deck as a dev environment is fraught since it has a read only filesystem making normal development tasks much more complicated. If you look at deck.sh you can see I had to specifically install pip into userspace to the deck then install packages to userspace with that pip.

Hopefully this addresses your questions.
